### ISSUE TITLE
db cleared with WAL mode

### ISSUE THREAD

**Issue by wmertens** [sqlitebrowser/sqlitebrowser#1353]
### Details for the issue

Sometimes, when I open a DB that is in WAL mode and was written to by a process that then quit before it closed the db, the DB is cleared after opening by DB4S. It is not always reproducible. I was able to reproduce it once on the latest nightly, but not any more.

I have the DB files that triggered it once, I can provide them privately.

Reproduction:
* create DB with WAL mode, add some things to it
* quit program without closing
* open with DB4S
* sometimes you get an empty screen, and the DB is also empty on disk.

#### I'm opening this issue because:

- [ ] DB4S is crashing
- [x] DB4S has a bug
- [ ] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [x] Mac OS: ( _version:_ High Sierra )
- [ ] Other: ___

#### I'm using DB4S version:

- [ ] 3.10.1
- [ ] 3.10.0
- [ ] 3.9.1
- [x] Other: Version 3.10.99 (Dec 11 2017)

#### I have also:

- [x] Tried out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds - But I couldn't reproduce it
- [x] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Oops.  Sorry @wmertens, somehow we seem to have missed this. 😦 

This kind of sounds familiar to me.  If you're able to trigger this again, or perhaps just keep an eye out for it... if/when it happens again, please check if the .wal file is left on disk right next to the database.

If it is, then (after backing up both) try deleting the .wal file.  I have a faint recollection that our startup code doesn't always handle things well when a .wal is already present, and might be causing the database to appear empty.  With the .wal file deleted first though, DB4S should display what's in the SQLite database file.

Note that's all from (dodgy) memory though. :wink:

**Comment by wmertens** [comment]
Hmm I think the dbs were actually cleared, WAL or no :( I'll keep an eye out some more, but actually I don't understand why DB4S would be doing anything with the .wal files, isn't all that sqlite's domain?

**Comment by justinclift** [comment]
Yeah, this is just something from memory of a behaviour I've already seen when doing "weird stuff" with existing sqlite files.  I don't remember the intricate details of what/where/how/why DB4S was acting weirdly though. :wink:
